thirdeye/thirdeye-frontend/app/pods/rootcause-anomalyfunction-cache/service.js

```diff
@@ -0,0 +1,126 @@
+import Service from '@ember/service';
+import {
+  toFilters,
+  toFilterMap
+} from 'thirdeye-frontend/utils/rca-utils';
+import { checkStatus } from 'thirdeye-frontend/utils/utils';
+import fetch from 'fetch';
+import _ from 'lodash';
+import moment from 'moment';
+
+export default Service.extend({
+  timeseries: null, // {}
+
+  context: null, // {}
+
+  pending: null, // Set
+
+  errors: null, // Set({ urn, error })
+
+  init() {
+    this.setProperties({ timeseries: {}, context: {}, pending: new Set(), errors: new Set() });
+  },
+
+  clearErrors() {
+    this.setProperties({ errors: new Set() });
+  },
+
+  request(requestContext, urns) {
+    const { context, timeseries, pending } = this.getProperties('context', 'timeseries', 'pending');
+
+    const metrics = [...urns].filter(urn => urn.startsWith('frontend:anomalyfunction:'));
+
+    // TODO eviction on cache size limit
+
+    let missing;
+    let newPending;
+    let newTimeseries;
+    if(!_.isEqual(context, requestContext)) {
+      // new analysis range: evict all, reload, keep stale copy of incoming
+      missing = metrics;
+      newPending = new Set(metrics);
+      newTimeseries = metrics.filter(urn => urn in timeseries).reduce((agg, urn) => { agg[urn] = timeseries[urn]; return agg; }, {});
+
+    } else {
+      // same context: load missing
+      missing = metrics.filter(urn => !(urn in timeseries) && !pending.has(urn));
+      newPending = new Set([...pending].concat(missing));
+      newTimeseries = timeseries;
+    }
+
+    this.setProperties({ context: _.cloneDeep(requestContext), timeseries: newTimeseries, pending: newPending });
+
+    if (_.isEmpty(missing)) {
+      // console.log('rootcauseAnomalyFunctionService: request: all metrics up-to-date. ignoring.');
```

Comment:
Do you need to keep this?

Reply:
I could turn this into a "regular" comment, but a comment there will be